spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -510,22 +508,13 @@ function chanUpgradeTry(
         (currentTimestamp() > timeout.timeoutTimestamp && timeout.timeoutTimestamp != 0)
     )
 
-    // if OPEN, then initialize handshake with upgradeFields
-    // otherwise, assert that the upgrade fields are the same for crossing-hellos case
-    if currentChannel.state == OPEN {
-        // if the counterparty sequence is greater than the current sequence, we fast forward to the counterparty sequence
-        // so that both channel ends are using the same sequence for the current upgrade
-        // initUpgradeChannelHandshake will increment the sequence so after that call
-        // both sides will have the same upgradeSequence
-        if counterpartyUpgradeSequence > currentChannel.upgradeSequence {
-            currentChannel.upgradeSequence = counterpartyUpgradeSequence - 1
-        }
+    // if the counterparty sequence is greater than the current sequence, we fast forward to the counterparty sequence
+    // so that both channel ends are using the same sequence for the current upgrade
+    // initUpgradeChannelHandshake will increment the sequence so after that call
+    // both sides will have the same upgradeSequence
+    abortTransactionUnless(counterpartyUpgradeSequence > currentChannel.upgradeSequence)
```

Comment:
This seems incorrect? code doc does not match the check and the check is using an unverified counterparty upgrade sequence, in addition it doesn't allow for the sequence to be equal?